### PR TITLE
Added Vision Transformer (ViT) Model

### PR DESCRIPTION
**Pull request by guglielmocamporese**
### Model info:
* Paper of ViT [[here](https://arxiv.org/pdf/2010.11929.pdf)]

### Usage: 
```python
from extra.vit import VisionTransformer

net = VisionTransformer(img_size=224, patch_size=16, in_chans=3, 
                        num_classes=10, embed_dim=128, depth=2, 
                        num_heads=4)
x = Tensor.uniform(1, 3, 224, 224, requires_grad=False)  
y = net.forward(x) # shape [1, 10]
```

### Test
```console
PYTHONPATH="." python3 examples/train_vit.py
```

### Main changes:
* added `extra/vit.py` where ViT is defined,
* added `example/train_vit.py` where ViT is benchmarked, for now only on MNIST, sorry,
* added layers to `tinygrad/nn`: `nn.Linear`, `nn.Conv2d`, `nn.Identity`, `nn.Sequential`, `nn.Dropout`,
* added `Cat` on `tinygrad/ops_cpu.py` for concatenating tensors that require gradients.

### CODE REVIEW

**Comment by geohot** [comment]
Let's get ResNet merged first. The "Cat" layer here requires some thought, as that would have to be written for all the "cpu, gpu, npe, ..."
